### PR TITLE
KTOR-7315 Deprecate old Jetty and Tomcat engines

### PR DESCRIPTION
**Pull request by bjhham**
**Subsystem**
Server, Jetty + Tomcat engines

**Motivation**
[KTOR-7315](https://youtrack.jetbrains.com/issue/KTOR-7315) Deprecate old jetty / tomcat / servlet engines

**Solution**
Add deprecation annotations to produce warnings.

### CODE REVIEW

File: ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/EngineMain.kt [hunk 1]
```diff
@@ -10,6 +10,7 @@ import io.ktor.server.engine.*
 /**
  * Jetty engine
  */
+@Deprecated("This references an old version of Jetty. Use ktor-server-jetty-jakarta module instead")
```
Comment (e5l) [inline ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/EngineMain.kt:13]: "Please consider changing dependency from 'ktor-server-jetty' to 'ktor-server-jetty-jakarta'"

**Review by e5l — APPROVED** [review]
@vnikolova, could you also check?

File: ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/EngineMain.kt (hunk 1 above)
Comment (vnikolova) [reply ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/EngineMain.kt:13]: I suggest the following:
"The `ktor-server-jetty` module is deprecated and will be removed in the next major release as it references an outdated version of Jetty. Please use the `ktor-server-jetty-jakarta` module instead."

Comment (e5l) [reply ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/EngineMain.kt:13]: We can replace next major release with `4.0.0` explicitly

**Pull request merged**